lib/urldata.h

```diff
@@ -1797,6 +1797,7 @@ struct UserDefined {
   BIT(crlf);            /* convert crlf on ftp upload(?) */
   BIT(strip_path_slash); /* strip off initial slash from path */
   BIT(ssh_compression);            /* enable SSH compression */
+  long haproxyprotocol; /* whether to send HAProxy PROXY protocol header */
```

Comment:
1. You could save a few bytes by not storing this as a `long`.
2. Move this field declaration to above the `BIT()` fields to better use struct space.